src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -339,6 +341,74 @@ public UserDetails retrieveUser(String username, String password, String domainN
         }
     }
 
+    /**
+     * Escape the DN escaped characters per http://social.technet.microsoft.com/wiki/contents/articles/5312.active-directory-characters-to-escape.aspx
+     *
+     * @param distinguishedName
+     *      The distinguisedName without escaping the ten characters plus the forward slash
+     * @return never null
+     */
+    public static String escapeCharactersInDistinguisedName (String distinguishedName, boolean flagForwardSlash) {
+        StringBuilder sb = new StringBuilder();
+        char[] chardistinguishedArray = distinguishedName.toCharArray();
+        for (int i = 0; i < distinguishedName.length(); i++) {
+            if (i > 3 && (i + 3) < distinguishedName.length()) {
+                String element = new StringBuilder().append(chardistinguishedArray[i + 1]).append(chardistinguishedArray[i + 2]).append(chardistinguishedArray[i + 3]).toString();
+                //If the next element is is CN, OU or DC we don't need to escape the comma that there is right before
+                if (!element.equals("CN=") && !element.equals("OU=") && !element.equals("DC=")) {
+                    //Characters to escape are documented in http://social.technet.microsoft.com/wiki/contents/articles/5312.active-directory-characters-to-escape.aspx
+                    switch (distinguishedName.charAt(i)) {
+                        case ',':
+                            sb.append("\\,");
+                            break;
+                        case '\\':
+                            sb.append("\\\\");
+                            break;
+                        case '#':
+                            sb.append("\\#");
+                            break;
+                        case '+':
+                            sb.append("\\+");
+                            break;
+                        case '<':
+                            sb.append("\\<");
+                            break;
+                        case '>':
+                            sb.append("\\>");
+                            break;
+                        case ';':
+                            sb.append("\\;");
+                            break;
+                        case '"':
+                            sb.append("\"");
+                            break;
+                        case '=':
+                            sb.append("\\=");
+                            break;
+                        case ' ':
+                            sb.append("\\ ");
```

Comment:
this is incorrect - the Space **only** needs escaping if it is the [leading or trailing](http://social.technet.microsoft.com/wiki/contents/articles/5312.active-directory-characters-to-escape.aspx) part of the DN.

> The space character must be escaped only if it is the leading or trailing character in any component of a distinguished name